Add Cloud-Native "motivation" to group goals (#12)
cncf/cnf-wg: charter.md

The CNF WG operates under the aegis of CNCF. The charter of the working group is to define the process around certifying the cloud nativeness of networking applications, aka CNFs. We collaborate with the [CNF test suite project](https://github.com/cncf/cnf-conformance/blob/master/README-testsuite.md) who works on the mechanics of the conformance tests.

> The goal for the group is to create a software conformance program that any application or network function implementation can use to demonstrate that they are conformant and interoperable with cloud native principles.

Reviewer: Note: This was already deleted. This PR is not against current charter.md

> - To build consensus around the benefits of using cloud native architecture methodologies for telecommunications.

Reviewer 2: ```suggestion
- To build consensus around the benefits of using cloud native architecture methodologies for telecommunications through user stories and use cases.
```

Reviewer: I think adding specific work items/deliverables like this would be good under the charters [In Scope](https://github.com/cncf/cnf-wg/blob/f6bd97dbcf145d2b1ad6ebf2e27da3d5538f2c84/charter.md#in-scope) section. The high-level goal could stay general rather than specifying the method for achieving it.

> - To build consensus around the benefits of using cloud native architecture methodologies for telecommunications.

Reviewer: This was a goal for the TUG and other efforts.  It seems like a repeat in the CNF WG. It is a sub-goal of every best practice (eg. what are the benefits of this best practice).

The primary audience for the CNF WG is for people that want to *gain* benefits and are looking for practices, methodologies, etc which will provide them.

If the goal is going to be listed it should be related to the best practices goal and come after it.  Something like:

```suggestion
- To highlight the benefits of using cloud native best practices for telecommunications
```

> - To build concensus around the motivation for using Cloud-Native architecture and DevOps methodologies for telecommunications.

Reviewer 3: I feel like this ship has sailed. Motivated or not, this technology shift is coming. Vendors are not going to provide offers that don't come packaged this way in the future. I think showing how cloud native practices map to service provider business outcomes would help. Up time is our numero uno. Pointing to self-healing services is an example on how this shift meets the architectural constraints of CSPs.

Author: Well, not so fast. I believe it is still a good drill to start with the motivation. You mentioned modern packaging and uptime. Some network functions (as you probably witnessed yourself) are just ported from VM's to Docker images, and continue using their 1+1 (on n+m) redundancy schemes, claiming they are "cloud-native". We want to avoid exactly that, and point to the tenets of cloud-native that provide the proper functionality. Also, I heard different cloud-native motivations from some of the other operators, like expecting vendors to provide smaller units of functionality, so they can integrate their own network functions using best of breed components. Not saying I subscribe to this approach, but  it exists anyway. Starting from agreeing on the motivation will, IMHO, keep this effort focused.

Reviewer: @rannyh while I think it will be good for the best practices which are discussed to include the motivations and benefits for the telecommunication industry I think it should not be stated as a primary goal.  Please see the [TUG whitepaper](https://github.com/cncf/telecom-user-group/blob/master/whitepaper/CNCF%20_TUG_Cloud%20Native%20Thinking%20for%20Telecommunications.pdf), or the multitude of books, blog posts, and videos for motivations in using cloud native and devops methodologies.

That said, [in scope for the CNF WG](https://github.com/cncf/cnf-wg/blob/master/charter.md#in-scope), is "Publishing metrics/white papers". So if a paper relating the cloud native benefits in telco terms for motivation would be beneficial that can be done.

It is not the goal / focus of this group though.

The primary goal is to help folks who want to obtain the benefits of cloud native principles and methodologies in gaining them.

Reviewer 3: @rannyh I don't think we are saying different things. Specifically, I am saying that I think the "do I go cloud-native?" at the macro level is sort of behind us. Individual motivations as to whether we go down path "x" or "y" based on the cloudiness of something is a different matter. That in turn ties to those business drivers. I feel like none of us would be here in these numbers if there wasn't already strong motivation to migrate towards a more cloud-native approach in general.

Reviewer: @rannyh I agree that having some context on the motivations will help.  I am saying it is secondary to more primary goals such as does a best practice provide enough context to communicate its benefits to creators, vendors, and operators.

We can have papers in the repo that help with context, add links and short text into the README though.

Author: @Taylor and @jeffsaelens - I am OK with not making "defining motivation" part of the goals. I never meant it to be an answer to "why go cloud-native?" but rather some sort of consensus on "which of the cloud-native tenets are relevant to CSPs?". I do see a situation where we may have a hard time agreeing on some "best practice" down the road. Having an agreed set of motivations could help resolve such debates. Maybe it is already covered well enough in the TUG Whitepaper and we should not repeat it here.

Reviewer: @rannyh I think that if someone proposes a best practice for a telco audience (dev, ops, etc) then the proposal should have the motivation and benefits listed. There should be references for why it's a best practice. Its part of the process and templates now.  If a best practice is not relevant to CSPs then that should come out in the discussion of a proposal.

Then in the meetings we can discuss proposals that have been put forth or bring up new ones.  More time to discuss how a best practice provides benefits to CSPs and any issues and alternatives.

Reviewer 3: I've been calling these requirements in my head. Does best practice "A" align with a CSP (or any network operator) requirement. This could have been misreading on my part, but it seemed a deep and bottomless pit to fall down in to if we were debating if we should be doing cloud-native-y things in the first place.

Reviewer: > Does best practice "A" align with a CSP (or any network operator) requirement. 

And what requirement does this best practice help fulfill.

Reviewer 3: when I think of uptime, I'm usually specifically referring to, "is the service reachable?". We can hide how the sausage is made from the consumer of the service in most instances. "Does my phone have 4g? yep! we're good here."

> - To build concensus around the motivation for using Cloud-Native architecture and DevOps methodologies for telecommunications.

Reviewer 2: ```suggestion
- To build consensus around the benefits of using cloud native architecture methodologies for telecommunications.
```

> - To create a software conformance program that any application or network function implementation can use to demonstrate that they are conformant and interoperable with cloud native principles.

Reviewer 3: Sounded like we are dropping conformance for now and moving to best practices. Maybe we shift the focus? "To provide a repository of best practices for CNF developers and operators alike, allowing them to demonstrate that the software and the operation of said function strives to adhere to cloud native principles."?

Reviewer: @jeffsaelens good catch. We do want to update the language to focus on best practices.

How about something like this:
- To create a program which identifies best practices for CNFs running on K8s which CNF Developers and operators alike may use to demonstrate how well their software and its operation adhere to cloud native principles.
- To determine which practices will allow telco applications to most efficiently utilize K8s to lower the operational risk and burdens of service providers and telco vendors

> - To create a software conformance program that any application or network function implementation can use to demonstrate that they are conformant and interoperable with cloud native principles.

Reviewer 2: ```suggestion
- To create a program that identifies best practices for CNFs running on K8s which CNF Developers and operators alike may use to demonstrate how well their software and its operations adhere to cloud native principles.
```
